src/base/tests/test_util.py

```diff
@@ -1481,6 +1486,27 @@ def test_iter(self):
         self.assertEqual(result, expected)
 
 
+class TestQueryIds(UnitTestCase):
+    def test_straight(self):
+        result = list(util.query_ids(self.env.cr, "SELECT * FROM (VALUES (1), (2)) AS x(x)", itersize=2))
+        self.assertEqual(result, [1, 2])
+
+    def test_chunks(self):
+        with util.query_ids(self.env.cr, "SELECT * FROM (VALUES (1), (2)) AS x(x)") as ids:
+            result = list(util.chunks(ids, 100, fmt=list))
+        self.assertEqual(result, [[1, 2]])
+
+    def test_destructor(self):
+        ids = util.query_ids(self.env.cr, "SELECT id from res_users")
+        del ids
+
+    def test_pk_violation(self):
+        with db_connect(self.env.cr.dbname).cursor() as cr, mute_logger("odoo.sql_db"), self.assertRaises(
+            ValueError
+        ), util.query_ids(cr, "SELECT * FROM (VALUES (1), (1)) AS x(x)") as ids:
+            list(ids)
```

Comment:
Add an explicit `del ids` to be sure uninitialized data doesn't break `del` when the PK error is raised.
```suggestion
            list(ids)
      del ids 
```
I'd add another test with an invalid query then `del obj` to double down in del not breaking even if the query is invalid. 

I know the exceptions here would stop the upgrade, but it looks odd to have ignored exceptions in `__del__` when the GC does the cleanup.

Reply:
> I'd add another test with an invalid query then `del obj`

Would that add anything? Adding `del obj` immediately before end-of-scope of `obj` will not make it being garbage collected earlier, will it?
IOW, the current test already does test it, doesn't it?